runtime/protocol/http-client-engines/http-client-engine-ktor/common/src/aws/smithy/kotlin/runtime/http/engine/ktor/KtorEngine.kt

```diff
@@ -121,7 +122,12 @@ class KtorEngine(
                 waiter.signal()
             }
 
-            val body = KtorHttpBody(httpResp.contentLength(), content)
+            val contentLength = httpResp.contentLength()
+            val body = if (contentLength == 0L && content.isClosedForRead) {
```

Comment:
**Question**: Out of curiosity, why is `content.isClosedForRead` necessary for the empty body condition? What does it mean if `contentLength == 0L` and `content.isClosedForRead == false`?